core/templates/dev/head/domain/editor/undo_redo/UndoRedoService.js

```diff
@@ -146,3 +146,127 @@ oppia.factory('UndoRedoService', [
     return UndoRedoService;
   }
 ]);
+
+// TODO(tjiang11): Figure out way to create new instance of service without
```

Comment:
I was just thinking on reading this that a lot of this code seems pretty general. Why can't we reuse the existing UndoRedoService instead of duplicating?

I'm concerned that this becomes one of those TODOs that never gets done (e.g. #4968 just appeared in my inbox) so I feel like we should do it properly since we'll need it for other things too? In general code duplication should be avoided as much as possible (and this is really quite a lot of it).

Reply:
Deduplicated